pravega-provisioner/cluster_provisioner.py

```diff
@@ -161,25 +183,40 @@ def resource_based_provisioning(vms, vm_cpus, vm_ram_gb, vm_local_drives, zookee
-    print("--------- Segment Store In-Memory Cache Size (Pravega +0.7) ---------")
-    print("Segment Store pod memory limit: ", Constants.segment_store_jvm_size_in_gb + new_direct_memory, "GB")
-    print("Segment Store JVM Size (-Xmx JVM Option) : ", Constants.segment_store_jvm_size_in_gb, "GB")
-    print("Segment Store direct memory (-DirectMemory JVM Option): ", new_direct_memory, "GB")
-    # We leave 1GB extra of direct memory for non-caching purposes
-    print("Segment Store cache size (pravegaservice.cacheMaxSize): ", (new_direct_memory - 1) * 1024 * 1024 * 1024,
-          " (", new_direct_memory - 1, "GB)")
-    print("Buffering time that Pravega tolerates with Tier 2 unavailable given a (well distributed) write workload:")
-    for w in range(100, 1000, 200):
-        print("- Write throughput: ", w, "(MBps) -> buffering time: ", ((segment_stores * (new_direct_memory - 1) * 1024) / w), " seconds")
+    new_direct_memory = 0 
+    if max_segment_stores_per_vm > 0:
+        new_direct_memory = Constants.segment_store_direct_memory_in_gb + int(min_vm_mem_available / max_segment_stores_per_vm)
+        print("--------- Segment Store In-Memory Cache Size (Pravega +0.7) ---------")
+        print("Segment Store pod memory limit: ", Constants.segment_store_jvm_size_in_gb + new_direct_memory, "GB")
+        print("Segment Store JVM Size (-Xmx JVM Option) : ", Constants.segment_store_jvm_size_in_gb, "GB")
+        print("Segment Store direct memory (-DirectMemory JVM Option): ", new_direct_memory, "GB")
+        # We leave 1GB extra of direct memory for non-caching purposes
+        print("Segment Store cache size (pravegaservice.cacheMaxSize): ", (new_direct_memory - 1) * 1024 * 1024 * 1024,
+              " (", new_direct_memory - 1, "GB)")
+        print("Buffering time that Pravega tolerates with Tier 2 unavailable given a (well distributed) write workload:")
+        for w in range(100, 1000, 200):
+            print("- Write throughput: ", w, "(MBps) -> buffering time: ", str(round((segment_stores * (new_direct_memory - 1) * 1024) / w, 2)), " seconds")
 
     # Add a warning if the number of Bookies is higher than the number of nodes, as Pravega may need to enable rack-
     # aware placement in Bookkeeper so it tries to write to Bookies in different nodes (data availability).
     if bookkeeper_servers > vms:
         print("WARNING: To guarantee data availability and durability, consider enabling rack-aware placement in "
               "Pravega to write to Bookkeeper.")
 
-    return zookeeper_servers, bookkeeper_servers, segment_stores, controllers
+    # zookeeper_servers, bookkeeper_servers, segment_stores, controllers
+    # return values in a dictionary instead of relying on long list of positional output values
+    # this also allows in adding any new values if required
+    output = dict()
+    output['controller'] = int(controllers)
```

Comment:
Please, if you are adding the output, please add all the fields (not only the ones that are good for the use that you want to do of this script). Please, this is an open source project, so let's think on the completeness of the code and genera use. For instance, why you are not adding the number of nodes/vms here?

Reply:
added other variables to output, please see above.